components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java

```diff
@@ -201,8 +200,9 @@ public void init(Properties properties) {
                 addConfigProperty(name, value);
             }
         }
-        log.debug("EI lightweight registry is initialized.");
+        log.debug("MI lightweight registry is initialized.");
 
+        initSecurityRepo();
```

Comment:
Whats the reason for moving this?

Reply:
Previously, it was not considering the locations that are configured in the registry.xml. It always tries to create the secure-vault folder in the default _MI_HOME/registry/_ location. It will cause issue in a read-only file system.